src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -90,19 +98,19 @@ public void enqueue(KafkaMessage message) {
 		}
 	}
 
-	@Override
 	public void start() {
 		this.running = true;
 	}
 
-	@Override
-	public void stop() {
+	public void stop(long stopTimeout) {
+		shutdownLatch = new CountDownLatch(1);
 		this.running = false;
```

Comment:
Yes, I see by your code.
What is the benefit of the change afterward?
Consider we use hands off `MessageChannel`.
Maybe an issue with the adapter's `phase` ?

Would you mind sharing the StackTrace in the JIRA?